zebra-chain/src/transaction/serialize.rs

```diff
@@ -491,6 +499,16 @@ impl ZcashDeserialize for Transaction {
                 if id != TX_V5_VERSION_GROUP_ID {
                     return Err(SerializationError::Parse("expected TX_V5_VERSION_GROUP_ID"));
                 }
+                // convert the nConsensusBranchId to a NetworkUpgrade
+                let network_upgrade =
+                    match NetworkUpgrade::network_upgrade(reader.read_u32::<LittleEndian>()?) {
```

Comment:
You could use `ok_or(SerializationError…)?` to simplify this code:
https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or